### PR TITLE
feat: Add Windows PowerShell support to CLI runtime

### LINKED ISSUE #9210: CLI mode does not fully support Windows+PowerShell
## Problem

The OpenHands CLI mode currently does not properly support Windows systems with PowerShell. While OpenHands has comprehensive PowerShell support implemented in `openhands/runtime/utils/windows_bash.py`, the CLI runtime (`openhands/runtime/impl/cli/cli_runtime.py`) only uses subprocess with bash commands, which may not work correctly on Windows systems.

## Current Behavior

- CLI mode hardcodes the use of bash commands via subprocess
- Windows users may experience issues when running commands through the CLI
- PowerShell support exists but is only used in the action execution server, not in CLI runtime

## Expected Behavior

- CLI mode should automatically detect Windows systems and use PowerShell instead of bash
- Windows users should have a seamless experience using OpenHands CLI

## Technical Details

- PowerShell support is already implemented in `openhands/runtime/utils/windows_bash.py`
- The action execution server already detects Windows with `sys.platform == 'win32'` and uses PowerShell
- CLI runtime needs similar detection and PowerShell integration

## Proposed Solution

Modify the CLI runtime to:
1. Detect Windows systems using `sys.platform == 'win32'`
2. Use the existing PowerShell implementation instead of bash subprocess calls
3. Ensure seamless cross-platform compatibility

Labels: bug, enhancement, windows, cli

### PR DESCRIPTION
**Pull request by neubig**
## Summary

This PR implements automatic Windows PowerShell support for the OpenHands CLI runtime, addressing issue #9210. The implementation leverages the existing `WindowsPowershellSession` infrastructure to provide better Windows compatibility.

### Core Implementation
- **Automatic Windows Detection**: CLI runtime now automatically detects Windows platform and enables PowerShell support
- **PowerShell Integration**: Integrates existing `WindowsPowershellSession` for command execution on Windows
- **Graceful Fallback**: Maintains backward compatibility with subprocess fallback when PowerShell is unavailable
- **Type Safety**: Proper type annotations and mypy compliance

### Key Features
- Platform-specific command execution (PowerShell on Windows, subprocess elsewhere)
- PowerShell session lifecycle management (initialization, execution, cleanup)
- Error handling and logging for PowerShell operations
- Seamless integration with existing CLI runtime functionality

### Testing
- Added comprehensive test plan (`WINDOWS_POWERSHELL_TEST_PLAN.md`) for Windows validation
- Includes cloud VM setup instructions and detailed test scenarios
- Covers functionality, performance, and regression testing

## Technical Details

The implementation adds:
1. Windows platform detection in CLI runtime initialization
2. PowerShell session management with proper lifecycle handling
3. Command execution method that routes to PowerShell on Windows
4. Error handling and fallback mechanisms
5. Type-safe imports using `TYPE_CHECKING`

## Testing Plan

The included test plan covers:
- Basic PowerShell command execution
- File operations and directory navigation
- Environment variable handling
- Error scenarios and edge cases
- Performance benchmarking
- Integration with OpenHands features

## Backward Compatibility

- No breaking changes to existing functionality
- Non-Windows platforms remain unaffected
- Graceful fallback when PowerShell is unavailable
- All existing CLI runtime features preserved

## Related Issues

Fixes #9210 - CLI mode doesn't completely support Windows+PowerShell

## Future Enhancements

This implementation provides the foundation for:
- Enhanced Windows-specific command support
- Better PowerShell cmdlet integration
- Improved Windows development experience
- Potential PowerShell-specific features

## Testing Checklist

- [x] Code compiles without errors
- [x] Pre-commit hooks pass (mypy, ruff, formatting)
- [x] Type annotations are correct
- [x] Error handling is comprehensive
- [ ] Manual testing on Windows VM (requires Windows environment)
- [ ] Integration testing with OpenHands features
- [ ] Performance benchmarking

The manual testing on Windows will be performed using the provided test plan once a Windows environment is available.

@neubig can click here to [continue refining the PR](https://app.all-hands.dev/conversations/093a6ca151ad418aad9717f0e0f631f9)

---

To run this PR locally, use the following command:
```
docker run -it --rm   -p 3000:3000   -v /var/run/docker.sock:/var/run/docker.sock   --add-host host.docker.internal:host-gateway   -e SANDBOX_RUNTIME_CONTAINER_IMAGE=docker.all-hands.dev/all-hands-ai/runtime:9f70454-nikolaik   --name openhands-app-9f70454   docker.all-hands.dev/all-hands-ai/openhands:9f70454
```

### CODE REVIEW

**Comment by neubig** [comment]
@openhands

I got the following error when trying to run the PowerShell runtime. I suspect that it is necessary to do a bit more installation on Windows in order to prevent the coreclr error.

Take a look at the windows without wsl guide, and create a new CLI section that contains *only* the necessary steps to install the CLI. It can include the command for install uvx on windows (if not already installed):

```
powershell -ExecutionPolicy ByPass -c "irm https://astral.sh/uv/install.ps1 | iex"
```

then the command to install coreclr, then the command to run OpenHands with uvx.

```
uvx --python 3.12 --from openhands-ai openhands
```

File: openhands/runtime/impl/cli/cli_runtime.py [hunk 1]
```diff
@@ -241,6 +289,48 @@ def _safe_terminate_process(self, process_obj, signal_to_send=signal.SIGTERM):
...
+
+            # Execute the command using the PowerShell session
+            result = self._powershell_session.execute(ps_action)
+
+            # The PowerShell session should return a CmdOutputObservation or ErrorObservation
+            if isinstance(result, CmdOutputObservation):
+                return result
+            else:
+                # This covers ErrorObservation and any other unexpected types
+                return result
```
Comment (neubig) [inline openhands/runtime/impl/cli/cli_runtime.py]:
```suggestion
            # Execute the command using the PowerShell session
            return self._powershell_session.execute(ps_action)
```

File: openhands/runtime/impl/cli/cli_runtime.py [hunk 2]
```diff
@@ -135,6 +169,20 @@ async def connect(self) -> None:
...
+                self._powershell_session = WindowsPowershellSession(
+                    work_dir=self._workspace_path,
+                    username=None,  # Use current user
+                    no_change_timeout_seconds=30,
+                    max_memory_mb=None,
+                )
+            except Exception as e:
+                logger.error(f'Failed to initialize PowerShell session: {e}')
+                logger.warning('Falling back to subprocess for command execution.')
+                self._powershell_session = None
```
Comment (neubig) [inline openhands/runtime/impl/cli/cli_runtime.py]:
```suggestion
           self._powershell_session = WindowsPowershellSession(
               work_dir=self._workspace_path,
               username=None,  # Use current user
               no_change_timeout_seconds=30,
               max_memory_mb=None,
           )
```

**Comment by neubig** [comment]
@openhands run pre-commit to fix linting

**Review by li-boxuan — APPROVED** [review]
The basic functionalities work for me, thanks!

There's a bigger problem, which is more related to the CLI itself, though: the CLI itself seems to eat my keystrokes randomly, making the CLI not quite usable. I don't think this problem existed in the old CLI.

**Comment by neubig** [comment]
I have also tested this and it works OK!
